### PR TITLE
Two classes for optimizing concentric circles have been added to radial.py.

### PR DESCRIPTION
**Pull request by carbonscott**
It follows PR #90, but I cannot push it to this PR directly for permission reasons.

### CODE REVIEW

**Comment by apeck12** [comment]
@carbonscott, thanks for these classes. I'll integrate them into the radial.py and modify the `AgBehenate` class accordingly and keep you posted about whether the optimization is working. One additional request: for the docstrings, would you also mind adding what the Returns for each function are?

**Comment by carbonscott** [comment]
@apeck12 I've provide what functions return in their doc strings.

**Comment by apeck12** [comment]
@carbonscott, thanks!

**Comment by carbonscott** [comment]
@apeck12 @fredericpoitevin I've made some changes to run a minimal refinement script.  The following plot shows a better overlap between peaks detected and peaks predicted for experiment `mfxp19619` run 19.  

r0019

**Review by apeck12 — APPROVED** [review]
@carbonscott, thanks for these commits!

File: btx/diagnostics/ag_behenate.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def opt_center(self, peaks_observed):
             radii of detected powder peaks in pixels
         """
         # Create a concentric circle model...
-        cx, cy = center
+        cx, cy = self.centers[-1]
         r = peaks_observed
         num = 200
```
[the remark this answers is not in the document]
Comment (carbonscott) [reply btx/diagnostics/ag_behenate.py:134]: @apeck12 Sorry for the hard coding.  It's better to make `num` a user-specified parameter.  I don't have strong evidence to conclude what is a good number for it.  However, sampling 200 pixels from each concentric circle (there are 18 circles/peaks detected and used for optimization) takes 40+ sec to optimize in experiment `mfxp19619` run `19`.

[the remark this answers is not in the document]
Comment (apeck12) [reply btx/diagnostics/ag_behenate.py:134]: Thanks for that detail. I'll set a reasonable default value after looking at a few other powders but make it an optional argument that can be reset by calling the `opt_geom` function.

[the remark this answers is not in the document]
Comment (carbonscott) [reply btx/diagnostics/ag_behenate.py:134]: @apeck12 I think we can use 200 as the default value.  We will find out a more reasonable value for `num` after a series of testing.

[the remark this answers is not in the document]
Comment (carbonscott) [reply btx/diagnostics/ag_behenate.py:134]: @apeck12 There's some room to improve it, but let's keep it for the next version.  The idea is that we can sample more pixels from larger circle, and vice versa.  The current version samples equal number of pixels, e.g. 200, irrespective of the circle size.

**Comment by carbonscott** [comment]
@apeck12 @fredericpoitevin One quick thing, the config file in `yaml` still uses `opt_distance`, do you want to change it to `opt_geom`?  The last commit is about changing the key from `opt_distance` to `opt_geom`.  However, if you have made a different decision, please ignore this commit.

**Comment by apeck12** [comment]
Thanks @carbonscott for noting that. I need to make a few additional updates to that task so will fix the associated nomenclature in the yaml files as well.

**Pull request closed without merging**